be/src/runtime/routine_load/routine_load_task_executor.cpp

```diff
@@ -273,34 +275,33 @@ void RoutineLoadTaskExecutor::exec_task(StreamLoadContext* ctx, DataConsumerPool
     switch (ctx->load_src_type) {
     case TLoadSourceType::KAFKA: {
         std::shared_ptr<DataConsumer> consumer;
-        Status st = _data_consumer_pool.get_consumer(ctx, &consumer);
+        Status st = consumer_pool->get_consumer(ctx, &consumer);
         if (!st.ok()) {
             // Kafka Offset Commit is idempotent, Failure should not block the normal process
             // So just print a warning
             LOG(WARNING) << st.get_error_msg();
             break;
         }
+        DeferOp return_consumer([&consumer_pool, &consumer] { consumer_pool->return_consumer(consumer); });
 
         std::vector<RdKafka::TopicPartition*> topic_partitions;
         for (auto& kv : ctx->kafka_info->cmt_offset) {
             RdKafka::TopicPartition* tp1 = RdKafka::TopicPartition::create(ctx->kafka_info->topic, kv.first, kv.second);
             topic_partitions.push_back(tp1);
         }
+        // delete TopicPartition finally
+        auto tp_deleter = [&topic_partitions]() {
+            std::for_each(topic_partitions.begin(), topic_partitions.end(),
+                          [](RdKafka::TopicPartition* tp1) { delete tp1; });
+        };
```

Comment:
I think the DeferOp is suitable for the instance which will be deconstructed or handled in many return places.
For the topic_partitions, there is no necessarily to define a DeferOp.